qemu/tests/virtio_fs_supp_group_transfer.py

```diff
@@ -0,0 +1,106 @@
+
+    try:
+        shared_dir = params.get("fs_source_dir")
+
+        if setup_local_nfs:
+            # delete the slash at the end
+            params["nfs_mount_dir"] = shared_dir[:-1]
+            nfs_local = nfs.Nfs(params)
+            nfs_local.setup()
+
+        params["start_vm"] = "yes"
+        env_process.preprocess_vm(test, params, env, params["main_vm"])
+        vm = env.get_vm(params.get("main_vm"))
+        vm.verify_alive()
+        guest_root_session = vm.wait_for_login()
+
+        error_context.context("Create a destination directory %s "
+                              "inside guest." % fs_dest, test.log.info)
+        if not utils_misc.make_dirs(fs_dest, session=guest_root_session):
+            test.fail("Creating directory was failed!")
+        error_context.context("Mount virtiofs target %s to %s inside"
+                              " guest." % (fs_target, fs_dest),
+                              test.log.info)
+        if not utils_disk.mount(fs_target, fs_dest, 'virtiofs',
+                                session=guest_root_session):
+            test.fail('Mount virtiofs target failed.')
+
+        error_context.context("Create a common user...", test.log.info)
+        guest_root_session.cmd(add_user_cmd % username)
+        error_context.context("Change the group to wheel", test.log.info)
+        guest_root_session.cmd("usermod -G wheel %s" % username)
+
+        error_context.context("Create a dir inside the virtiofs and "
+                              "change it's group to wheel", test.log.info)
+        guest_root_session.cmd("cd %s && mkdir -m 770 %s" % (fs_dest, testdir))
+        guest_root_session.cmd("cd %s && chgrp wheel %s" % (fs_dest, testdir))
+        params["fs_dest"] = os.path.join(fs_dest, testdir)
+        params["fs_source_dir"] = os.path.join(shared_dir, testdir)
+
+        error_context.context("Login the common user and try to write under "
+                              "the dir which belongs to wheel group",
+                              test.log.info)
+        guest_user_session = vm.wait_for_login()
+        guest_user_session.cmd("su %s" % username)
+        virtio_fs_utils.basic_io_test(test, params, guest_user_session)
+    finally:
+        if guest_root_session:
+            output = guest_root_session.cmd_output(del_user_cmd % username)
+            if "is currently used by process" in output:
+                error_context.context("Kill process before delete user...",
+                                      test.log.info)
+                pid = output.split(" ")[-1]
+                guest_root_session.cmd_output("kill -9 %s" % pid)
+            guest_root_session.cmd("rm -rf /home/%s" % username)
+
+            guest_root_session.cmd("rm -rf %s" % params["fs_dest"])
+
+        if setup_local_nfs:
+            if vm and vm.is_alive():
+                vm.destroy()
+            params["rm_mount_dir"] = params.get("nfs_mount_dir")
+            nfs_local.cleanup()
```

Comment:
```suggestion
            if nfs_local:
                nfs_local.cleanup()
```
As your design, better to check if the `nfs_local` is None before `cleanup()`, otherwise, I think there is a logical leak.